monai/data/utils.py

```diff
@@ -1067,17 +1066,17 @@ def compute_importance_map(
     if mode == BlendMode.CONSTANT:
         importance_map = torch.ones(patch_size, device=device, dtype=torch.float)
     elif mode == BlendMode.GAUSSIAN:
-        center_coords = [i // 2 for i in patch_size]
+
         sigma_scale = ensure_tuple_rep(sigma_scale, len(patch_size))
         sigmas = [i * sigma_s for i, sigma_s in zip(patch_size, sigma_scale)]
 
-        importance_map = torch.zeros(patch_size, device=device)
-        importance_map[tuple(center_coords)] = 1
-        pt_gaussian = GaussianFilter(len(patch_size), sigmas).to(device=device, dtype=torch.float)
-        importance_map = pt_gaussian(importance_map.unsqueeze(0).unsqueeze(0))
-        importance_map = importance_map.squeeze(0).squeeze(0)
-        importance_map = importance_map / torch.max(importance_map)
-        importance_map = importance_map.float()
+        importance_map = 0
+        for i in range(len(patch_size)):
+            x = torch.arange(start=-(patch_size[i] - 1) / 2.0, end=(patch_size[i] - 1) / 2.0 + 1, dtype=torch.double)
```

Comment:
Can you use https://github.com/Project-MONAI/MONAI/blob/4082b351f5dea1848703ec8365b12b0346bacbe2/monai/networks/layers/convutils.py#L80 the implementation here is incorrect

Reply:
>the implementation here is incorrect

which imlementation is incorrect ??

> Can you use 

I don't think we should use it.  it creates a spaghetti code. We  want to allocate a  simple smooth map, why do we need so many dependencies.   that function gaussian_1d has many options (for truncations, for approximations, etc), and we need to import additional modules

Reply:
The device and dtype handling may cause unecessary overheads

Reply:
alright, I moved calc on gpu right away, see below

Reply:
> I don't think we should use it. it creates a spaghetti code. We want to allocate a simple smooth map, why do we need so many dependencies. that function gaussian_1d has many options (for truncations, for approximations, etc), and we need to import additional modules

Hi @myron, the functions I suggested are monai utility functions based on pytorch API. these don't introduce new dependencies.
https://github.com/Project-MONAI/MONAI/blob/4ad790cc5f6022a92e0585c34f2f50fb6fa7fd62/monai/networks/layers/convutils.py#L80

Let's reuse the utility function, or improve the utility function if the quality is not good enough. In this way we build a set of common APIs together.

In this specific case, if the logic indeed becomes complicated in the future, we can make a separate `gaussian_1d_sampled` and call this function within `gaussian_1d`

```py
def gaussian_1d(...):
   ...
   elif approx.lower() == "sampled":
   gaussian_1d_sampled(...)

def gaussian_1d_sampled(...):
   ...
```
so that the user can choose to use `gaussian_1d` or `gaussian_1d_sampled`.

for the same reason I made the comment here https://github.com/Project-MONAI/MONAI/pull/5281#discussion_r990419684 please also consider updating that PR.

Reply:
@wyli  Are you asking to replace this 2 lines of code 
```
x = x = torch.arange(start=-(patch_size[i] - 1) / 2.0, end=(patch_size[i] - 1) / 2.0 + 1, dtype=torch.float, device=device)
x = torch.exp(x**2 / (-2 * sigmas[i] ** 2))  
```
with  1 line of code?
```
x = gaussian_1d(sigma=patch_size[i] - 1) / 2.0, truncated=1, approx="sampled", normalize= False)
```

I see no reason for it:
- gaussian_1d does not work with even size patch_size (which is what we usually use for inference)  - gaussian_1d will always return an odd length results (https://github.com/Project-MONAI/MONAI/blob/4ad790cc5f6022a92e0585c34f2f50fb6fa7fd62/monai/networks/layers/convutils.py#L117)
- it does not take patch_size as an input, you'll need to put it in sigma (as a workaround)
- it has too many other options , e.g. trucated that a we need to rememeber to set to 1, to get the right result, which can lead to bugs in the future

for me to fix it I'll need write another function (that works with even length array), and then unit tests for it.  I see no reason to over-complicate this just to replace 2 lines with 1 line of code.  There is a simple formula for 1d gaussian, and we can just use it.  Our API convenience functions must simplify the code, and not to make it more convoluted.

Reply:
it should be changed because in the future if we decided to use another way to implement the gaussian kernel (for example there are changes in torch APIs or we use jit or c++ extensions), we know how to maintain this utility function. instead of searching through all the functions and identifying variants of the implementation.

Reply:
I think, in the interest of time, let's approve it as is, not to block further improvements.  And later when someone wants to create a separate function based on those 2 lines, and there is a real need for it, we can have another PR.  Currently the code is functioning well, is faster, and fixes the bug, so I think it's OK to merge it. Thank you

Reply:
ok, I'll merge this one for now, given that the speed up is useful...